src/event_pusher/mod_event_pusher_hook_translator.erl

```diff
@@ -22,83 +22,97 @@
-         unacknowledged_message/2]).
-
--ignore_xref([filter_local_packet/1, unacknowledged_message/2, user_not_present/5,
-              user_present/2, user_send_packet/4]).
+-export([user_send_packet/3,
+         filter_local_packet/3,
+         user_present/3,
+         user_not_present/3,
+         unacknowledged_message/3]).
 
 %%--------------------------------------------------------------------
 %% gen_mod API
 %%--------------------------------------------------------------------
 
 -spec add_hooks(mongooseim:host_type()) -> ok.
 add_hooks(HostType) ->
-    ejabberd_hooks:add(hooks(HostType)).
+    gen_hook:add_handlers(hooks(HostType)).
 
 -spec delete_hooks(mongooseim:host_type()) -> ok.
 delete_hooks(HostType) ->
-    ejabberd_hooks:delete(hooks(HostType)).
+    gen_hook:delete_handlers(hooks(HostType)).
 
 %%--------------------------------------------------------------------
 %% Hook callbacks
 %%--------------------------------------------------------------------
 -type routing_data() :: {jid:jid(), jid:jid(), mongoose_acc:t(), exml:element()}.
--spec filter_local_packet(drop) -> drop;
-                         (routing_data()) -> routing_data().
-filter_local_packet(drop) ->
-    drop;
-filter_local_packet({From, To, Acc0, Packet}) ->
+-spec filter_local_packet(drop, _, _) -> {ok, drop};
+                         (routing_data(), _, _) -> {ok, routing_data()}.
+filter_local_packet(drop, _, _) ->
+    {ok, drop};
+filter_local_packet({From, To, Acc0, Packet}, _, _) ->
     Acc = case chat_type(Acc0) of
               false -> Acc0;
-              Type ->
-                  Event = #chat_event{type = Type, direction = out,
-                                      from = From, to = To, packet = Packet},
-                  NewAcc = mod_event_pusher:push_event(Acc0, Event),
-                  merge_acc(Acc0, NewAcc)
+              Type -> push_chat_event(Acc0, Type, {From, To, Packet}, out)
           end,
-    {From, To, Acc, Packet}.
-
--spec user_send_packet(mongoose_acc:t(), From :: jid:jid(), To :: jid:jid(),
-                       Packet :: exml:element()) -> mongoose_acc:t().
-user_send_packet(Acc, From, To, Packet = #xmlel{name = <<"message">>}) ->
-    case chat_type(Acc) of
-        false -> Acc;
-        Type ->
-            Event = #chat_event{type = Type, direction = in,
-                                from = From, to = To, packet = Packet},
-            NewAcc = mod_event_pusher:push_event(Acc, Event),
-            merge_acc(Acc, NewAcc)
-    end;
-user_send_packet(Acc, _From, _To, _Packet) ->
-    Acc.
-
--spec user_present(mongoose_acc:t(), UserJID :: jid:jid()) -> mongoose_acc:t().
-user_present(Acc, #jid{} = UserJID) ->
+    {ok, {From, To, Acc, Packet}}.
+
+-spec user_send_packet(Acc, Args, Extra) -> {ok, Acc} when
+      Acc :: mongoose_acc:t(),
+      Args :: map(),
+      Extra :: map().
+user_send_packet(Acc, _, _) ->
+    Packet = mongoose_acc:packet(Acc),
+    ChatType = chat_type(Acc),
+    ResultAcc = if
```

Comment:
As for me it's fine, but don't be surprised if other devs ask you to replace `if` construction with `case`.